Automattic/jetpack
# Add recommendation for VideoPress

#### Changes proposed in this Pull Request:

* This PR adds a new recommendation for Jetpack VideoPress that shows 2 weeks after a site connects if VideoPress is not already enabled on that site.

#### Jetpack product discussion


pbNhbs-1iw-p2

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






To reset the assistant state while testing, you can use the following commands ( you can run these with `jetpack docker` if using the Docker development environment )
* `wp jetpack options delete recommendations_data`
* `wp jetpack options delete recommendations_conditional`
* `wp jetpack options delete recommendations_step`

If your test site already has a Jetpack VideoPress product, or if the VideoPress module is already enabled, you will not receive the new recommendation as outlined in the test plan below. Try testing using a site that has a Jetpack free plan and the VideoPress module disabled.

This PR is easiest to test on a local Jetpack site as there's a need to temporarily change some code.

---

* Checkout this branch on your Jetpack test site.
* Visit the Jetpack dashboard. Note that the "Recommendations" tab in the dashboard navigation does not show any badge indicating that there is a new recommendation.
![Screen Shot 2022-03-21 at 2 27 58 PM](https://user-images.githubusercontent.com/18016357/159340092-2adb1a7e-7834-4986-b965-33519fa1ee11.png)

* Now, go to the Jetpack Dashboard.
* Under "Site Connection", click on "Manage Site Connection" and disconnect Jetpack.
* After disconnecting Jetpack, locate the `jetpack_connected` method in the `Jetpack_Recommendations` class. Temporarily modify the following line: 
```
$date->add( new DateInterval( 'P14D' ) );
```
To be:
```
$date->add( new DateInterval( 'PT1M' ) );
```
* This will attempt to enable the VideoPress recommendation one minute after the site reconnects.
* Now, re-connect Jetpack. After a minute, visit a page on the front-end of the site.
* Navigate back to the Jetpack Dashboard. You should see a red badge with a "1" next to the Recommendations tab in the navigation on the Jetpack dashboard - indicating a new recommendation is available.
![Screen Shot 2022-03-11 at 4 00 23 PM](https://user-images.githubusercontent.com/18016357/158703217-3b7bb9bb-a1e9-4b6a-814e-d81e167094f1.png)

* Click on the "Recommendations" tab in the navigation - you should now see a recommendation step that provides some information about VideoPress. There should be a green "New" badge that shows on this step in the upper-left corner. The CTA button should enable the VideoPress module and take you to the summary screen. Alternatively, you can click on "Decide Later" to reach the summary screen. 
![Screen Shot 2022-03-30 at 2 44 06 PM](https://user-images.githubusercontent.com/18016357/160908633-66589588-867c-4c9e-9315-43f2cb77e96a.png)

* On the summary screen, you should see "VideoPress" in the list of Recommended Features. If not enabled, the "Enable" button should enable the VideoPress module. If already enabled, the CTA should link to the following support page: https://jetpack.com/support/jetpack-videopress/add-video-block-editor/
![Screen Shot 2022-03-30 at 2 45 03 PM](https://user-images.githubusercontent.com/18016357/160908685-5b3e8672-516f-45f5-8c24-340cb68ba5e9.png)

* If you reload the page on the summary screen, the green "New" badge next to VideoPress in the features list and the red "1" indicator next to "Recommendations" in the dashboard navigation  should no longer show.
* Test that the CTA buttons on both the individual recommendation and on the summary screen enable the VideoPress module.
* Automated unit tests should be passing
* Visuals should match this reference: G151IYl8Lm4f0NjgLpR943-fi-2132%3A39570
